src/transform-inline/visitor-type-check.ts

```diff
@@ -8,6 +8,83 @@ import * as VisitorIsStringKeyof from './visitor-is-string-keyof';
 import * as VisitorTypeName from './visitor-type-name';
 import { sliceSet } from './utils';
 
+function visitDateType(type: ts.ObjectType, visitorContext: VisitorContext) {
+    const name = VisitorTypeName.visitType(type, visitorContext, {type: 'type-check'});
+    return VisitorUtils.setFunctionIfNotExists(name, visitorContext, () => {
+        return ts.createFunctionDeclaration(
+            undefined,
+            undefined,
+            undefined,
+            name,
+            undefined,
+            [ts.createParameter(undefined, undefined, undefined, VisitorUtils.objectIdentifier, undefined, undefined, undefined)],
+            undefined,
+            ts.createBlock(
+                [ts.createIf(
+                    ts.createPrefix(
+                        ts.SyntaxKind.ExclamationToken,
+                        ts.createParen(ts.createBinary(
+                            ts.createIdentifier('object'),
+                            ts.createToken(ts.SyntaxKind.InstanceOfKeyword),
+                            ts.createPropertyAccess(
+                                ts.createIdentifier('global'),
```

Comment:
This will work for NodeJS apps, but in the browser `global` should be `window`.
If you leave out `global`/`window` it'd work on both?

Reply:
I wouldn't leave it out (to avoid conflicts with other classes of the same name) but I could add a check if `global` is available and test against `window` if it ins't. How about that?

Reply:
I suppose my proposal won't work because previously defined custom `class Date` will trigger a false positive.
I guess a polyfill will need to be added for `globalThis` which can then be used.
https://blog.logrocket.com/what-is-globalthis-why-use-it/

Reply:
Adding a check sounds good. How will you achieve it? If I try to reference `global` in the browser I get a `ReferenceError`

Reply:
I have not yet tried i but my best guess at the moment would be: try to reference it, use it if it works or use window it that get's us the `ReferenceError`.

Reply:
I've found nicer a better way that doesn't require to trigger an error:
```javascript
if (typeof global === 'undefined')
    // use window
else
    // use global
```